src/app/app.component.ts

```diff
@@ -82,7 +88,21 @@ export class MyApp {
     // Reset the content nav to have just this page
     // we wouldn't want the back button to show in this scenario
     this.menuCtrl.close().then(() => {
-      this.nav.setRoot(page.component, { 'linkParam': page.param });
+      if(page.component === PushMessagesPage && !PushService.registered) {
```

Comment:
@pawelpaszki @wei-lee Do we want to show this error message if the PushService fails to register? For example, when running on an iOS simulator the PushService will always fail to register, but that doesn't necessarily mean the configuration is missing.

Reply:
@aidenkeating I am open to any suggestions. I have implemented what I thought is required. Unfortunately I am still waiting on my mac to arrive and using lenovo laptop with fedora, so I can't test the app running in iOS. @wei-lee if you want me to change the behaviour, please let me know

Reply:
@pawelpaszki @aidenkeating let's keep it the same across different platforms and we only care about the configuration for the moment. We can still show the push messages page even if the push service is not registered.

Reply:
@wei-lee so are we only going to check for missing keycloack configuration?

Reply:
@pawelpaszki we should also check if there is push configuration missing? But we don't need to check if push service is registered.

Reply:
@wei-lee @pawelpaszki On iOS native that is what we do.

* If the configuration is missing we show the dialog when users try to navigate to the Push Messages page
* If registration fails we still allow users to navigate to the Push Messages page.

Reply:
The acceptance criteria say if the service is not configured, are we to assume configured in this sense means configured in the app (config in mobile-service.json) not the service has been provisioned properly and the app can interact with the service.

Reply:
@ciaranRoche That is my understanding, that configured means that the services configuration exists in `mobile-services.json` and is valid.